db/install.xml

```diff
@@ -1,28 +1,36 @@
+        <FIELD NAME="keepuserid" TYPE="int" LENGTH="10" NOTNULL="false" SEQUENCE="false" COMMENT="User that will remain with data of both users."/>
         <FIELD NAME="mergedbyuserid" TYPE="int" LENGTH="10" NOTNULL="false" SEQUENCE="false"/>
-        <FIELD NAME="timemodified" TYPE="int" LENGTH="10" NOTNULL="true" SEQUENCE="false" COMMENT="Time stamp when the merging action took place."/>
-        <FIELD NAME="log" TYPE="text" NOTNULL="true" SEQUENCE="false" COMMENT="List of actions performed if success, or errors on merging failure."/>
+        <FIELD NAME="timeadded" TYPE="int" LENGTH="10" NOTNULL="true" SEQUENCE="false" COMMENT="Time stamp when the merging request was added to the queue."/>
+        <FIELD NAME="timemodified" TYPE="int" LENGTH="10" NOTNULL="false" SEQUENCE="false" COMMENT="Time stamp when the merging request status has beeen updated."/>
+        <FIELD NAME="status" TYPE="int" LENGTH="4" NOTNULL="true" SEQUENCE="false" COMMENT="Was the action concluded ok?"/>
+        <FIELD NAME="retries" TYPE="int" LENGTH="4" NOTNULL="true" SEQUENCE="false" COMMENT="Number of retries to merge user accounts."/>
+        <FIELD NAME="log" TYPE="text" NOTNULL="false" SEQUENCE="false" COMMENT="List of actions performed if success, or errors on merging failure."/>
       </FIELDS>
       <KEYS>
         <KEY NAME="primary" TYPE="primary" FIELDS="id"/>
+        <KEY NAME="removeuserid" TYPE="foreign" FIELDS="removeuserid" REFTABLE="user" REFFIELDS="id"/>
+        <KEY NAME="keepuserid" TYPE="foreign" FIELDS="keepuserid" REFTABLE="user" REFFIELDS="id"/>
       </KEYS>
       <INDEXES>
-        <INDEX NAME="mdl_toolmerg_tou_ix" UNIQUE="false" FIELDS="touserid" COMMENT="index for touserid field." NEXT="mdl_toolmerg_fru_ix"/>
-        <INDEX NAME="mdl_toolmerg_fru_ix" UNIQUE="false" FIELDS="fromuserid" COMMENT="index for fromuserid field." PREVIOUS="mdl_toolmerg_tou_ix" NEXT="mdl_toolmerg_suc_ix"/>
-        <INDEX NAME="mdl_toolmerg_suc_ix" UNIQUE="false" FIELDS="success" COMMENT="Was the merging action ok?" PREVIOUS="mdl_toolmerg_fru_ix" NEXT="mdl_toolmerg_tfs_ix"/>
-        <INDEX NAME="mdl_toolmerg_tfs_ix" UNIQUE="false" FIELDS="touserid, fromuserid, success" COMMENT="Index enabling fast looking for merging actions for given pair of users." PREVIOUS="mdl_toolmerg_suc_ix"/>
-      </INDEXES>
+        <INDEX NAME="mdl_toolmergqueu_rem_ix" UNIQUE="false" FIELDS="removeuservalue" COMMENT="Index for removeuservalue field."/>
+        <INDEX NAME="mdl_toolmergqueu_kee_ix" UNIQUE="false" FIELDS="keepuservalue" COMMENT="Index for fromuserid field."/>
+        <INDEX NAME="mdl_toolmergqueu_timsta_ix" UNIQUE="false" FIELDS="timeadded, status" COMMENT="When listing the queue you probably would like to see the last updated or the last created to see how it went. Without indexes, that listing would be very expensive."/>
```

Comment:
These indexed maybe have to be thought again when we build the list of request on the web. I mentally thought about some scenarios and these indices maybe are not so convenient. We will see.